### ISSUE TITLE
Move annotation attributes into separate package

### ISSUE THREAD

**Issue by cd21h** [AArnott/MessagePack-CSharp#31]
Goal is to have lightweight package with annotations only which can be easily distributed with API contracts assembly.

Annotations to move to new assembly:
* `IgnoreMemberAttribute`
* `KeyAttribute`
* `MessagePackFormatterAttribute`
* `MessagePackObjectAttribute`
* `SerializationConstructorAttribute`
* TBD

**Comment by AArnott** [comment]
Seems reasonable to me.

**Comment by cd21h** [comment]
Done in #32

**Comment by AArnott** [comment]
@shatl I may revert this change. Having two assemblies instead of one is raising the startup time in Visual Studio, which is consuming the new 2.0 version. Given that in every condition that the attributes are used the full library will already be loaded, I don't know how a second assembly is beneficial.
If you can make the case why this separate assembly is important, please do so.

**Comment by cd21h** [comment]
@AArnott How much it affects startup time? 
Are you talking about Visual Studio opening project? I did not notice any significant VS startup time increase on my laptop. Unless it is minutes I would not care about this :)
Just out of curiosity - can you give a some numbers?

The reason for having separate assembly with attributes - 

When I ship nuget package with API contracts only (DTOs and API interface) for the service which supports content negotiation (msgpack, bson, json) I want to keep package as lightweight as possible and  leave decision which serializer to use up to client (e.g. some old clients may support json only).
Does this makes sense?

**Comment by AArnott** [comment]
You're probably not going to notice one assembly's difference in loading, but it adds up. And while on SSDs it's pretty fast, on HDDs (which many users still have) each individual file to be loaded requires another disk seek and that does quickly add up. 

Your use case makes sense, but if you're offering a data contract library and want to avoid bringing in all of MessagePack in case they don't want to use it, why use the MessagePack attributes at all? Why not use `[DataContract]` and `[DataMember]`?

**Comment by cd21h** [comment]
DataContract / DataMember does not provide all functionality which MsgPack attributes provide.

If we talking about application startup time, this is microseconds which happen only once. This is absolute
I would not sacrifice flexibility in favor of startup time improvement which may not even noticeable in many cases.

**Comment by AArnott** [comment]
What functionality do you need that you can't get through DataContract/DataMember?

**Comment by AArnott** [comment]
OK, never mind. I just looked at the msgpack attributes and I see there's quite a bit of options there that data contract does not expose.

**Comment by AArnott** [comment]
Can you open a new issue on the root project to track your request?

**Comment by cd21h** [comment]
Sure, https://github.com/neuecc/MessagePack-CSharp/issues/486 

The code is already in master

**Comment by AArnott** [comment]
> The code is already in master

This confused me, so I looked into it. Apparently neuecc#482 was incomplete in that it left the old library around although its attributes were no longer honored by the library and were meant to be deleted/moved instead of copied. I've fixed this (to delete the old copies). 

But of course your new issue is still active and I may "fix" it the other way yet.
